state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2109,6 +2112,23 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				},
 			};
 
+		if let Some(metadata) = deposit_metadata.clone() {
+			if T::CcmValidityChecker::check_and_decode(&metadata.channel_metadata, output_asset)
+				.is_err()
+			{
+				log::warn!("Failed to process vault swap due to invalid CCM metadata");
+				return;
+			}
+
+			let destination_chain: ForeignChain = output_asset.into();
+			if !destination_chain.ccm_support() {
+				log::warn!(
+					"Failed to process vault swap due to destination chain not supporting CCM"
+				);
+				return;
+			}
+		}
+
```

Comment:
I moved this code that checks CCM from the perform_channel action to here. As I see it we only need this check for vault swaps because for channel  swaps this is already checked when opening the deposit channel.